_posts/2018-08-28-announcing-riff-0-1-2-on-Knative.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: single
+title: "Announcing riff v0.1.2 on Knative"
+header:
+  overlay_image: /images/liverpool-st.jpg
+excerpt:
+  With verbose builds and function chaining using Channels and Subscriptions    
+permalink: /blog/announcing-riff-0-1-2-on-Knative/
+---
+
+[riff v0.1.2](https://github.com/projectriff/riff/releases/tag/v0.1.2) on Knative is now available. Many thanks to all riff and Knative contributors.
+
+## install
+We recommend installing riff v0.1.2 on a fresh Kubernetes cluster. The riff CLI can be downloaded from our [releases page](https://github.com/projectriff/riff/releases) on GitHub. Please follow one of the [getting started](/docs) guides,  to create a new cluster on GKE or minikube.
+
+## feedback during builds
+There are two new options to provide feedback during builds. In previous releases, `riff function create` would return immediately after creating the build resources, without waiting for the build to succeed or fail.
+
+- `--wait` or `-w` waits until the build status is known.
+- `--verbose` or `-v` is the same as wait, but also relays the logs from each build step to the CLI output.
+
+For example if you have not [initialized](/docs/getting-started-with-knative-riff-on-minikube/#initialize-the-namespace) the default namespace, creating a function with `--wait` will produce an error message.
+
+```
+riff function create node square \
+  --git-repo https://github.com/trisberg/node-fun-square.git \
+  --artifact square.js \
+  --image $DOCKER_ID/node-fun-square:v1 \
+  --wait
+```
+```
+Error: function creation failed: RevisionMissing: Configuration "square" does not have any ready Revision.; Revision creation failed with message: "Internal error occurred: admission webhook \"webhook.build.knative.dev\" denied the request: mutation failed: serviceaccounts \"riff-build\" not found".
+```
+
+Or, if the specified artifact does not exist, `--verbose` will help you figure this out.
```

Comment:
Hmmm. This gives the impression that `--wait` isn't sufficient, but I think `--wait` would be good enough. Is there another example where the end status doesn't give enough detail about what happened? One suggestion would be that `--verbose` lets you see _progress_ and gives you some insight into what's going on.

Reply:
That is indeed the case:
```
riff function create node square3 --git-repo https://github.com/trisberg/node-fun-square.git \
  --artifact squarexxx.js \
--image gcr.io/$GCP_PROJECT/will-fail:v1 --wait
Error: function creation failed: RevisionMissing: Configuration "square3" does not have any ready Revision.; Revision "square3-00001" failed with message: "build step \"build-step-build-and-push\" exited with code 1 (image: \"docker-pullable://gcr.io/kaniko-project/executor@sha256:345cb6294526735acd66d58e92f74584cd21289621ed3bdb0cc21bdeb51570d2\"); for logs run: kubectl -n default logs square3-00001-qbjv6 -c build-step-build-and-push".
```

vs

```
riff function create node square4 --git-repo https://github.com/trisberg/node-fun-square.git \
  --artifact squarexxx.js \
--image gcr.io/$GCP_PROJECT/will-fail:v1 --verbose
Waiting for LatestCreatedRevisionName:. square4-00001
....
default/square4-00001-5v88n[build-step-build-and-push]: Error: error building image: lstat /workspace/squarexxx.js: no such file or directory
...
Error: function creation failed: RevisionMissing: Configuration "square4" does not have any ready Revision.; Revision "square4-00001" failed with message: "build step \"build-step-build-and-push\" exited with code 1 (image: \"docker-pullable://gcr.io/kaniko-project/executor@sha256:345cb6294526735acd66d58e92f74584cd21289621ed3bdb0cc21bdeb51570d2\"); for logs run: kubectl -n default logs square4-00001-5v88n -c build-step-build-and-push".
```